packages/mcdoc/src/runtime/attribute/builtin.ts

```diff
@@ -272,7 +285,46 @@ export function registerBuiltinAttributes(meta: core.MetaRegistry) {
 					const error = message
 						.replace(/^Invalid regular expression: /, '')
 						.replace(/^\/.+\/: /, '')
-					ctx.err.report(localize('invalid-regex-pattern', error), node, 2)
+					ctx.err.report(
+						localize('invalid-regex-pattern', error),
+						node,
+						core.ErrorSeverity.Warning,
+					)
+				}
+			}
+		},
+	})
+	registerAttribute(meta, 'match_regex', regexMatchValidator, {
+		checker: (config, typeDef, _) => {
+			if (typeDef.kind !== 'literal' || typeDef.value.kind !== 'string') {
+				return undefined
+			}
+			const pattern = config.pattern
+			const value = typeDef.value.value
+			return (node, ctx) => {
+				try {
+					const regex = RegExp(pattern)
+					if (!regex.test(value)) {
+						if (config.message) {
+							ctx.err.report(config.message, node, core.ErrorSeverity.Warning)
+						} else {
+							ctx.err.report(
+								localize('mismatching-regex-pattern', typeDef.value.value),
```

Comment:
You can use `const message = config.message ?? localize('mismatching-regex-pattern', typeDef.value.value)` here to simplify this to a single `ctx.err.report` call